### PR TITLE
Add comments to proxy_agent.

### PR DESCRIPTION
**Pull request by ZhidongPeng**

### CODE REVIEW

File: proxy_agent/src/service/windows.rs [hunk 1]
```diff
@@ -1,6 +1,10 @@
 // Copyright (c) Microsoft Corporation
 // SPDX-License-Identifier: MIT
 
+//! This module contains the logic to start the GPA service as a Windows service, hook up stop service control handler.
+//! The GPA service is implemented as a Windows service using the windows_service crate.
+//! Tt is started, stopped, and controlled by the Windows service manager.
```
Comment (shahneerali) [inline proxy_agent/src/service/windows.rs]: what is "Tt"

File: proxy_agent/src/service/windows.rs [hunk 2]
```diff
@@ -1,6 +1,10 @@
 // Copyright (c) Microsoft Corporation
 // SPDX-License-Identifier: MIT
 
+//! This module contains the logic to start the GPA service as a Windows service, hook up stop service control handler.
```
Comment (shahneerali) [inline proxy_agent/src/service/windows.rs]: This module contains the logic to start the GPA service as a Windows service _and_ hook up stop service control handler.

File: proxy_agent/src/shared_state.rs [hunk 3]
```diff
@@ -19,48 +19,93 @@ use windows_service::service_control_handler::ServiceStatusHandle;
...
+
     // agent_status
+    /// The flag to indicate if the agent status module is shutdown
     agent_status_shutdown: bool,
+    /// The proxy connection summary from the proxy
     proxy_summary: HashMap<String, ProxyConnectionSummary>,
+    /// The failed authenticate summary from the proxy
     failed_authenticate_summary: HashMap<String, ProxyConnectionSummary>,
+
     // proxy
```
Comment (shahneerali) [inline proxy_agent/src/shared_state.rs:93]: would it be more concise to add "proxy" to the next line comment

File: proxy_agent/src/shared_state.rs [hunk 4]
```diff
@@ -139,12 +193,33 @@ pub mod shared_state_wrapper {
...
 /// wrapper functions for KeyKeeper related state fields
+/// Example:
+/// ```rust
+/// use proxy_agent::shared_state::key_keeper_wrapper;
+/// use proxy_agent::shared_state::SharedState;
+/// use std::sync::{Arc, Mutex};
+///
+/// let shared_state = SharedState::new();
+///
+/// // set the key once the MSP feature is enabled
```
Comment (shahneerali) [inline proxy_agent/src/shared_state.rs]: maybe define MSP feature here for clarity

Comment (ZhidongPeng) [reply proxy_agent/src/shared_state.rs]: removed msp

File: proxy_agent/src/service/windows.rs (hunk 2 above)
Comment (ZhidongPeng) [reply proxy_agent/src/service/windows.rs]: updated.

File: proxy_agent/src/service/windows.rs (hunk 1 above)
Comment (ZhidongPeng) [reply proxy_agent/src/service/windows.rs]: It

**Review by shahneerali — APPROVED** [review]

**Pull request merged**